snakefile.py

```diff
@@ -218,21 +242,35 @@ def tool(name):
 
 
 rule fastqc_raw:
-    input: os.path.join(READS_DIR, "{sample}_R{read_num}.fastq")
+    input: trim_reads_input
     output:
-        os.path.join(FASTQC_DIR, '{sample}', '{sample}_R{read_num}_fastqc.zip'),
-        os.path.join(FASTQC_DIR, '{sample}', '{sample}_R{read_num}_fastqc.html')
-    log: os.path.join(LOG_DIR, 'fastqc_{sample}_raw_R{read_num}.log')
+        r1_rep = os.path.join(FASTQC_DIR, '{sample}', '{sample}_R1_fastqc.html'),
+        r1_zip = os.path.join(FASTQC_DIR, '{sample}', '{sample}_R1_fastqc.zip'),
+        r2_rep = os.path.join(FASTQC_DIR, '{sample}', '{sample}_R2_fastqc.html'),
+        r2_zip = os.path.join(FASTQC_DIR, '{sample}', '{sample}_R2_fastqc.zip') # all outputs are provided to ensure atomicity 
+    log: [os.path.join(LOG_DIR, 'fastqc_{sample}_raw_R1.log'), os.path.join(LOG_DIR, 'fastqc_{sample}_raw_R2.log')]
     params:
         output_dir = os.path.join(FASTQC_DIR, '{sample}')
-    shell: "{FASTQC_EXEC} -o {params.output_dir} {input} >> {log} 2>&1"
+    run:
+        # renaming the ".fastq.gz" suffix to "_fastqc.html" 
+        tmp_R1_output = os.path.basename(input[0])[:-9] + '_fastqc.html'
+        tmp_R1_zip = os.path.basename(input[0])[:-9] + '_fastqc.zip'
+        tmp_R2_output = os.path.basename(input[1])[:-9] + '_fastqc.html'
+        tmp_R2_zip = os.path.basename(input[1])[:-9] + '_fastqc.zip'
+        shell("""{FASTQC_EXEC} -o {params.output_dir} {input} >> {log} 2>&1;
+                if [[ {tmp_R1_output} != *{wildcards.sample}* ]]; then
```

Comment:
I don't understand the meaning of this condition.  What's the goal here?

Reply:
When the sample name is the same as the read-files the file already has the correct name. 
`mv` throws an error when I try to rename the file to the name it already has. 

So I made this condition so that the renaming will only be executed if the sample name is not already in the read-file-tmp name